src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -50,24 +50,26 @@ namespace Xamarin.Android.Net
 	/// by AndroidClientHandler is requested by the server, the application can provide its own authentication module (<see cref="AuthenticationData"/>, 
 	/// <see cref="PreAuthenticationData"/>) to handle the protocol authorization.</para>
 	/// <para>AndroidClientHandler also supports requests to servers with "invalid" (e.g. self-signed) SSL certificates. Since this process is a bit convoluted using
-	/// the Java APIs, AndroidClientHandler defines two ways to handle the situation. First, easier, is to store the necessary certificates (either CA or server certificates)
-	/// in the <see cref="TrustedCerts"/> collection or, after deriving a custom class from AndroidClientHandler, by overriding one or more methods provided for this purpose
-	/// (<see cref="ConfigureTrustManagerFactory"/>, <see cref="ConfigureKeyManagerFactory"/> and <see cref="ConfigureKeyStore"/>). The former method should be sufficient
-	/// for most use cases, the latter allows the application to provide fully customized key store, trust manager and key manager, if needed. Note that the instance of
-	/// AndroidClientHandler configured to accept an "invalid" certificate from the particular server will most likely fail to validate certificates from other servers (even
-	/// if they use a certificate with a fully validated trust chain) unless you store the CA certificates from your Android system in <see cref="TrustedCerts"/> along with
-	/// the self-signed certificate(s).</para>
+	/// the Java APIs, AndroidClientHandler defines a way to handle the situation. It can store the necessary certificates (either CA or server certificates)
+	/// in the <see cref="TrustedCerts"/> collection. If, however, the application requires finer control over the SSL configuration (e.g. it implements its own
+	/// TrustManager) then it should derive a custom class from <see cref="Xamarin.Android.Net.AndroidMessageHandler"/> instead of using AndroidClientHandler.
+	/// Note that the instance of AndroidClientHandler configured to accept an "invalid" certificate from the particular server will most likely fail to validate
+	/// certificates from other servers (even if they use a certificate with a fully validated trust chain) unless you store the CA certificates from your Android
+	/// system in <see cref="TrustedCerts"/> along with the self-signed certificate(s).</para>
 	/// </remarks>
-	public class AndroidClientHandler : HttpClientHandler
+	[Obsolete("AndroidClientHandler has been deprecated. Use AndroidMessageHandler instead.")]
+	public sealed class AndroidClientHandler : HttpClientHandler
```

Comment:
I agree with the concern here: https://github.com/xamarin/xamarin-android/pull/6541#issuecomment-985045429

If this is `sealed`, how do you implement "ssl pinning" in an app?

Reply:
All the original logic was transferred into `AndroidMessageHandler` so anything that was previously possible to achieve by extending the `AndroidClientHandler` will be possible by extending `AndroidMessageHandler`.

Reply:
So if there is a NuGet package out there now that subclasses `AndroidClientHandler` internally, then updating Xamarin.Android they would probably run into a scary runtime exception?

@jonpryor can we break the API like this in Xamarin.Android?

It seems like this break should be reasonable in .NET 6, as customer would expect breaks as they move over.

Reply:
I think it would make sense to obsolete the type in .NET6 and keep things the same in Xamarin.Android.

Reply:
@simonrozsival I think we would wrap the `[Obsolete]` attribute with:
```csharp
#if !MONOANDROID1_0
[Obsolete("AndroidClientHandler has been deprecated. Use AndroidMessageHandler instead.")]
#endif
```

Reply:
@jonathanpeppers The attribute is only used in `AndroidClientHandler.cs` (included only when `TargetFramework` is `!= monoandroid10` and not in `AndroidClientHandler.Legacy.cs` which is included for `monoandroid10` (https://github.com/xamarin/xamarin-android/blob/main/src/Mono.Android/Mono.Android.csproj#L350..L351). Shouldn't that be enough?

Reply:
Missed that, thanks.